kntest/pkg/cluster/gke/README.md

```diff
@@ -0,0 +1,59 @@
+## kntest cluster gke
+
+`kntest cluster gke` command is used for creating, deleting or getting a GKE
+cluster
+
+## Prerequisite
+
+- `GOOGLE_APPLICATION_CREDENTIALS` set
+
+## Usage
+
+This tool can be invoked from command line with following parameters:
+
+- `--min-nodes`: minimum number of nodes, default 1
+- `--max-nodes`: maximum number of nodes, default 3
+- `--node-type`: GCE node type, default "e2-standard-4"
+- `--region`: GKE region, default "us-central1"
+- `--zone`: GKE zone, default empty
+- `--project`: GCP project, default empty
+- `--name`: cluster name, default empty
+- `--release-channel`: GKE release channel, default empty
+- `--resource-type`: Boskos resource type
+- `--version`: GKE version
+- `--backup-regions`: backup regions to be used if cluster creation in primary
```

Comment:
`--backup-regions` is only used when cluster creation in the main region fails, so in the normal scenario only `--region` will be used. I think `--region=xx,yy,zz` will be confusing as it would be like users want to create 3 clusters in the 3 regions.

Reply:
Then lets change it to "--possible-regions" or "--usable-regions" or "--candidate-regions"

Reply:
TBH I don't see why these are better than having `--region` and `--backup-regions`.. By separating them into two flags, users can just provide `--region` if they do not care which other regions are considered to be backups. But I assume changing to `--possible-regions` will require users to always provide multiple regions? Or only one region if they do not care about backups? Either way is more confusing I think.
Further more, we have `--zone` option, do we also want to change it to `--possible-zones`?

Reply:
I don't see why `--possible-regions` would not accept a single option.
Or `--region` could be a repeated option instead of having --backup-options:

```
--region us-central1
--region us-west1
```
...etc

Also, zone is confusing with multiple regions being possible. It might be best to have a repeated `--location` option that takes either region or zone:

```
--location us-central1-a
--location us-west1
```
etc

Reply:
I double checked the code, another reason we are doing like this is we are following how it's done in bash to make sure they are compatible - https://github.com/knative/test-infra/blob/master/scripts/e2e-tests.sh#L42-L48.

There are also other places tied to it that makes it hard to change, e.g. https://github.com/knative/pkg/blob/master/testutils/clustermanager/e2e-tests/setup.go#L91-L108.

Since this needs to change two repos, I would prefer to not make changes in this PR. I can clean them up as you suggested later for sure.

Reply:
If we're just recapitulating mess because that's the way it was in bash, then there's little point in doing this at all. 
I grepped through docs, serving, eventing, test-infra, and pkg, and none set zone ever.
Looking at `gcloud container clusters create`, it takes EITHER --zone or --region.
--region is e.g. us-central1
--zone is e.g. us-central1-a
Additionally, it supports `--node-locations` which allows nodes to be spread across zones in one region.
Supporting zones doesn't make any sense when we have backup regions.
If people really want all the knobs, then they can call `gcloud` directly.
For this case, let's streamline and make it more reliable. No zones, only regions, passed like so:
```
--region us-central1
--region us-west1
--region etc
```

Reply:
[cobra](https://github.com/spf13/cobra) does not seem to support flag like this. 

Also like I mentioned above, [`BackupRegions`](https://github.com/knative/test-infra/pull/1883/files#diff-9a42c6de599ac1363d68dd7b4e9d4e65R41) is one field of the request struct, it would be pretty dirty if I make changes in this PR without modifying `pkg` first.

I can clean them up and change to `--possible-regions` in followup PRs. But let's not block this PR because of this?

Reply:
It does and it's easy -- I'll send you a link directly.

They end up in a []string; you can slice so [1:] elements go to BackupRegions

Reply:
IMO backup regions should be considered a second class citizen, since it basically is a workaround, I would say that it shouldn't be mixed with the mainstream args. Also `--region a --region b` could be very confusing to users

Reply:
It would only be confusing if they don't know how to read.
It's not a workaround, it's a vital piece of functionality we rely on.